src/Uno.UI/UI/Xaml/Input/PointerRoutedEventArgs.wasm.cs

```diff
@@ -1,28 +1,73 @@
 ﻿using System;
+using System.Globalization;
+using System.Threading;
+using Windows.Devices.Input;
+using Windows.Foundation;
+using Windows.System;
 using Windows.UI.Input;
 
 namespace Windows.UI.Xaml.Input
 {
-	public partial class PointerRoutedEventArgs
+	partial class PointerRoutedEventArgs
 	{
+		private static long _pseudoNextFrameId;
+		private readonly uint _pseudoFrameId = (uint)Interlocked.Increment(ref _pseudoNextFrameId);
+		private readonly ulong _pseudoTimestamp = (ulong)DateTime.UtcNow.Ticks;
+
+		private readonly Point _absolutePosition;
+		private readonly VirtualKey _button;
+		private readonly PointerUpdateKind _updateKind;
+
+		internal PointerRoutedEventArgs(
+			uint pointerId,
+			PointerDeviceType pointerType,
+			Point absolutePosition,
+			bool isInContact,
+			VirtualKey button,
+			VirtualKeyModifiers keys,
+			PointerUpdateKind updateKind,
+			UIElement receiver)
+		{
+			_absolutePosition = absolutePosition;
+			_button = button;
+			_updateKind = updateKind;
+
+			Pointer = new Pointer(pointerId, pointerType, isInContact, isInRange: true);
+			KeyModifiers = keys;
+			OriginalSource = receiver; // This is not true, however we currently do not have a way to get it on WASM
```

Comment:
What's the value in this case ?

Reply:
It's the current "receiver" of the native event

Reply:
Finally I was able to get the real OriginalSource 🎉